rpcserver.go

```diff
@@ -3168,39 +3168,44 @@ func (r *rpcServer) sendPayment(stream *paymentStream) error {
 		htlcSema <- struct{}{}
 	}
 
+	// We keep track of the running goroutines and set up a quit signal we
+	// can use to request them to exit if the method returns because of an
+	// encountered error.
+	var wg sync.WaitGroup
+	reqQuit := make(chan struct{})
```

Comment:
This is a bit tricky and may not be obvious from the comment (reqQuit signalling quit on this execution line upon returning). I had to read thorugh the long function to find the one place where this might happen. Maybe add consider adding a comment there too.